kahuna/public/js/components/gr-undelete-image/gr-un-delete-image.js

```diff
@@ -0,0 +1,55 @@
+import angular from 'angular';
+import '../gr-confirm-delete/gr-confirm-delete';
+
+export const undeleteImage = angular.module('gr.undeleteImage', [
+    'gr.confirmDelete',
+    'util.async'
+]);
+
+undeleteImage.controller('grUnDeleteImageCtrl', [
+    '$rootScope', '$q', '$timeout', 'mediaApi', 'apiPoll',
+    function ($rootScope, $q, $timeout, mediaApi, apiPoll) {
+        var ctrl = this;
+
+        function pollDeleted (image) {
+            const findImage = () => mediaApi.find(image.data.id).then(
+                () => $q.reject(),
+                // resolve when image cannot be found, i.e. image has been deleted.
+                () => $q.resolve()
+            );
+
+            apiPoll(findImage);
+        }
+
+        ctrl.unDeleteImage = function (image) {
+            return mediaApi.undelete(image.data.id)
+                .then(() => pollDeleted(image))
+                .catch((err) => {
+                    $rootScope.$emit('image-delete-failure', err, image);
+                });
+        };
+
+        ctrl.undelete = function () {
+            // HACK to wait for thrall to process the message so that when we
+            // poll the api, it will be up to date.
```

Comment:
```suggestion
        ctrl.unDeleteSelected = function () {
```
if I understand this correctly, I wouldn't consider what you're doing here to be a 'hack', it's simply calling `unDeleteImage` on all the selected images, before sending the `images-deleted` event.